server/src/main/java/org/opensearch/index/engine/Engine.java

```diff
@@ -169,13 +169,30 @@ public final EngineConfig config() {
 
     protected abstract SegmentInfos getLastCommittedSegmentInfos();
 
+    /**
+     * Return the latest active SegmentInfos from the engine.
+     * @return {@link SegmentInfos}
+     */
+    protected abstract SegmentInfos getLatestSegmentInfos();
+
```

Comment:
Isn't this a breaking change for 2.x?

Reply:
Good call @Rishikesh1159 We'll need to provide default impl here and remove abstract.